lib/iopromise/data_loader.rb

```diff
@@ -3,13 +3,20 @@
 module IOPromise
   module DataLoader
     module ClassMethods
-      def attr_promised_data(*args)
+      def attr_async_data(attr_name, build_func = nil)
```

Comment:
**Naming** Actually, another naming suggestion I have here would be `async_attr`. The term `data` seems a bit weird, redundant? `async_attr` is what crossed my mind to get rid of it. What do you think?

Reply:
I agree the `data` suffix doesn't really make sense anymore but I slightly prefer `attr_async` over `async_attr`. The pattern of `attr_reader`, `attr_writer`, etc is more prominent than our internal `async_` pattern in my mind.

Reply:
This is a good point, let's change this before it's baked in everywhere 😀 I am strongly in favour of `attr_async` to keep `attr_` at the start to match up with the Ruby pattern for `attr_reader`, `attr_writer` and `attr_accessor` - but the fact that it's an "attribute" does definitely imply that it's data and not an action, since by definition it's exposing data via accessor methods, so `data` does seem redundant.

Reply:
Nice, glad I mentioned it then. 😄 I like `attr_async` better as well, consistency with the Ruby stdlib makes sense. 👍🏻